Update the dependencies of Dapr sample apps

ytimocin opened the pull request:
Fixing: #1895 

Updating the dependencies of the Dapr sample apps.

`dotnet run --project ui.csproj` is working locally. Now testing it in the pipeline.

ytimocin committed:
Updating Dapr UI Sample to dotnet 6

Signed-off-by: ytimocin <ytimocin@microsoft.com>

ytimocin commented on .devcontainer/post-create.sh:
Just a refactor of the file

ytimocin commented on samples/dapr/nodeapp/Dockerfile:
```diff
@@ -1,4 +1,4 @@
-FROM node:8-alpine
+FROM node:22-alpine
```
Update to node 22 from 8 (which is very old)

ytimocin commented on samples/dapr/nodeapp/app.js:
refactor

ytimocin commented on samples/dapr/nodeapp/package-lock.json:
Ran `npm update`

ytimocin commented on samples/dapr/nodeapp/package.json:
Updated the dependency (because I happened to be working in this folder)

ytimocin commented on samples/dapr/ui/Dockerfile:
```diff
@@ -1,15 +1,22 @@
-FROM mcr.microsoft.com/dotnet/sdk:5.0-alpine AS build
+FROM mcr.microsoft.com/dotnet/sdk:8.0-alpine AS build
```
dotnet 5 was too old :)

ytimocin commented on samples/dapr/ui/Dockerfile:
```diff
@@ -1,15 +1,22 @@
 WORKDIR /src
 
 COPY *.csproj .
-RUN dotnet restore -r linux-musl-x64
```
Removed the `linux-musl-x64`. I am still learning about Runtime Identifiers and Framework-Dependent Publishing: https://learn.microsoft.com/en-us/dotnet/core/deploying/.

sk593 replied:
Is this architecture what was causing the issue?

ytimocin commented on samples/dapr/ui/Dockerfile:
```diff
@@ -1,15 +1,22 @@
+FROM mcr.microsoft.com/dotnet/aspnet:8.0-alpine
 WORKDIR /app
+
+RUN addgroup -S appgroup && adduser -S appuser -G appgroup
```
Added this for extra security.

ytimocin commented on samples/dapr/ui/global.json:
I added this but not sure if this project needs it. I will do research on it.

ytimocin commented on samples/dapr/ui/ui.csproj:
```diff
@@ -1,11 +1,8 @@
-
   <ItemGroup>
-    <PackageReference Include="Dapr.Client" Version="1.1.2" />
+    <PackageReference Include="Dapr.Client" Version="1.14.0" />
```
Updated the Dapr.Client version.

ytimocin commented on samples/dapr/ui/ui.sln:
This got automatically generated.

sk593 replied:
What is this used for? If it gets automatically generated but it's not used anywhere do we need to track it in the repo? We could add it to the gitignore otherwise

kachawla reviewed (commented):
Just curious - how did we figure out that the issue is related to dependency versions?

ytimocin commented:
> Just curious - how did we figure out that the issue is related to dependency versions?

To be honest, I am not exactly sure. I built the image locally from the main branch and it wasn't working. There were also no logs in the Pod that gave me a clue. I decided to update the **Dapr UI App** from dotnet 5 which was too old to dotnet 8. I updated the **Dockerfile** accordingly. Then when I built an image that image started working fine for me.

I think it was also time that we updated the versions for the applications. The Node App for Dapr was using Node 8. I updated it to 22. And, as mentioned, UI App was using dotnet 5. That got updated to dotnet 8. For dependencies, we were using Dapr 1.2.0 which got updated to 1.14.0.
